### PR TITLE
move populateIpFromString to iceUtil. fix null terminating in iceAgen…

### PR DESCRIPTION
**Pull request by chehefen**
…tAddIceServer. Add More unit tests

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/source/Ice/IceAgent.c [hunk 1]
```diff
@@ -190,46 +192,15 @@ STATUS freeIceAgent(PIceAgent* ppIceAgent)
     return retStatus;
 }
 
-STATUS populateIpFromString(PIceCandidate pIceCandidate, PCHAR pBuff, UINT32 buffLen) {
```
Comment (MushMal) [inline src/source/Ice/IceAgent.c]: Name this populateIpV4FromString

File: src/source/Ice/IceAgent.c [hunk 2]
```diff
@@ -190,46 +192,15 @@ STATUS freeIceAgent(PIceAgent* ppIceAgent)
...
 STATUS iceAgentAddIceServer(PIceAgent pIceAgent, PCHAR url, PCHAR username, PCHAR credential) {
     ENTERS();
     STATUS retStatus = STATUS_SUCCESS;
     PIceServer pIceServer = NULL;
     PCHAR separator = NULL, urlNoPrefix = NULL;
     UINT32 port = ICE_STUN_DEFAULT_PORT;
 
-    CHK(url != NULL && username != NULL && credential != NULL, STATUS_NULL_ARG);
+    // only url is mandatory
+    CHK(url != NULL, STATUS_NULL_ARG);
```
Comment (MushMal) [inline src/source/Ice/IceAgent.c]: What about pIceAgent

File: src/source/Ice/IceUtils.c [hunk 3]
```diff
@@ -166,5 +166,40 @@ STATUS iceUtilsSendStunPacket(PStunPacket pStunPacket, PBYTE password, UINT32 pa
...
+
+    CHK(pKvsIpAddress != NULL && pBuff != NULL, STATUS_NULL_ARG);
+    CHK(STRNLEN(pBuff, KVS_MAX_IPV4_ADDRESS_STRING_LEN) > 0 &&
+        STRNLEN(pBuff, KVS_MAX_IPV4_ADDRESS_STRING_LEN) < KVS_MAX_IPV4_ADDRESS_STRING_LEN, STATUS_INVALID_ARG);
+
+    curr = pBuff;
+    tail = pBuff + STRLEN(pBuff);
+    while ((next = STRNCHR(curr, tail - curr, '.')) != NULL) {
+        CHK_STATUS(STRTOUI32(curr, curr + (next - curr), 10, &ipValue));
+        MEMCPY(pKvsIpAddress->address + octet, (PUINT8) &ipValue, SIZEOF(UINT8));
```
Comment (MushMal) [inline src/source/Ice/IceUtils.c]: Don't use memcpy here. Instead:

pKvsIpAddress->address[octet] = (UINT8) ipValue;

File: src/source/Ice/IceUtils.c [hunk 4]
```diff
@@ -166,5 +166,40 @@ STATUS iceUtilsSendStunPacket(PStunPacket pStunPacket, PBYTE password, UINT32 pa
...
+        octet++;
+
+        curr = next + 1;
+    }
+
+    CHK_STATUS(STRTOUI32(curr, NULL, 10, &ipValue));
+    MEMCPY(pKvsIpAddress->address + octet, (PUINT8) &ipValue, SIZEOF(UINT8));
```
Comment (MushMal) [inline src/source/Ice/IceUtils.c]: same

File: src/source/Ice/IceAgent.c (hunk 2 above)
Comment (chehefen) [reply src/source/Ice/IceAgent.c]: ops forgot

File: src/source/Ice/IceAgent.c (hunk 1 above)
Comment (chehefen) [reply src/source/Ice/IceAgent.c]: we may add ipv6 support in the future.

**Pull request merged**